src/marker.rs

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: © Contributors to the iceoryx-rs project
+// SPDX-FileContributor: Mathias Kraus
+
+/// # Safety
+///
+/// This is a marker trait for types that can be transferred via shared memory.
+/// The types must satisfy the following conditions:
+/// - no heap is used
+/// - the data structure is entirely contained in the shared memory - no pointers
+///   to process local memory, no references to process local constructs, no dynamic allocators
+/// - the data structure has to be relocatable and therefore must not internally
+///   use pointers/references
+/// - the type must not impl Drop; drop will not be called when the memory is released since the
+///   memory might be located in a shm segment without write access
+/// In general, types that could implement the Copy trait fulfill these requirements.
+pub unsafe trait ShmSend {}
+
+unsafe impl ShmSend for bool {}
+
+unsafe impl ShmSend for i8 {}
+unsafe impl ShmSend for u8 {}
+unsafe impl ShmSend for i16 {}
+unsafe impl ShmSend for u16 {}
+unsafe impl ShmSend for i32 {}
+unsafe impl ShmSend for u32 {}
+unsafe impl ShmSend for i64 {}
+unsafe impl ShmSend for u64 {}
+unsafe impl ShmSend for i128 {}
+unsafe impl ShmSend for u128 {}
+unsafe impl ShmSend for f32 {}
+unsafe impl ShmSend for f64 {}
+unsafe impl ShmSend for isize {}
+unsafe impl ShmSend for usize {}
+
+unsafe impl ShmSend for char {}
```

Comment:
@elBoberido what is your intended approach regarding support for the `iceoryx::cxx`  heap and exception free implementations of `std` types? Namely, `iox::cxx::string`.

Sorry, if this questions seems a bit out of context, so let me give you some context. I'm currently working on some `iceoryx` C++ <-> Rust interoperability use case. IMO, this is probably the most compelling use case for `iceoryx-rs` right now. 

I have a basic example, which works for integral types. Once I use a `string`, things fall apart, which is expected. From what I can tell, `Subscriber::getChunk()` is currently simply not prepared to work with non-integral types. Specifically [this](https://github.com/eclipse-iceoryx/iceoryx-rs/blob/275e6e0e4117b27842f741533c02782e2a7ab3dd/src/sb/ffi.rs#L260) line seems to be problematic for non-integral types.

Having some kind of support for `iox::cxx::string` will cover a lot of ground, I guess. Even if it is only `chars`, i.e. `u8`. WDYT?

Reply:
Another type which is often used would be `iox::cxx::vector`, so when we find a solution for the string we should also tackle the vector challenge at the same time when it makes sense.

Other containers like `stack`, `list` and `variant` are more corner cases but if they can be quickly supported we can do them as well.

Reply:
@adoerr the idea is to have something similar to this https://docs.rs/cxx/latest/cxx/#builtin-types

So there would be a `CxxString`, `IoxString` or `IoxCxxString` which would have the same layout as the C++ counterpart. What makes things interesting is the template argument but that should also be solvable.

Of course we would have to be really carefully with this and need to adapt the types when the ABI of the C++ counterpart changes